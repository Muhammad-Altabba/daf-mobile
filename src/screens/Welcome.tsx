/**
 * Serto Mobile App
 *
 */

import * as React from "react";
import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import Config from "react-native-config";
import Analytics from "appcenter-analytics";
import codePush from "react-native-code-push";

Analytics.setEnabled(true);

interface Props {}
interface State {
  updateMetadata: string | null;
}

export default class App extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      updateMetadata: ""
    };
  }
  onButtonPress() {
    codePush.sync({
      updateDialog: {
        appendReleaseDescription: true
      },
      installMode: codePush.InstallMode.IMMEDIATE
    });
  }

  componentDidMount() {
    codePush.getUpdateMetadata().then(updateMetadata => {
      this.setState({ updateMetadata: updateMetadata && updateMetadata.label });
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Welcome to updated2 Serto!</Text>
        <Text style={styles.welcome}>Environment = {Config.ENV}</Text>
        <TouchableOpacity
          onPress={() => {
            throw new Error("Sample error nr2");
          }}
        >
          <Text style={styles.welcome}>Press here to crash the app</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => {
            Analytics.trackEvent("Sample event");
          }}
        >
          <Text style={styles.welcome}>Press here to trigger sample event</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={this.onButtonPress}>
          <Text>Check for updates</Text>
        </TouchableOpacity>
        <Text>Update: {this.state.updateMetadata}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#F5FCFF"
  },
  welcome: {
    fontSize: 20,
    textAlign: "center",
    margin: 10,
    color: Config.BRAND_COLOR
  }
});
