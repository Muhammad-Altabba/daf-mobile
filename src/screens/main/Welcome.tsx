/**
 * Serto Mobile App
 *
 */
import React from 'react'
import {
  Container,
  Text,
  FabButton,
  Constants,
  Screen,
  Credential,
} from '@kancha/kancha-ui'
import { NavigationScreen } from '../../navigators'
import { Icons } from '../../theme'

const sertoVerifiableCredential = {
  iss: 'Serto Identity Platform',
  sub: 'did:ethr:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74',
  type: 'Serto ID',
  iat: 1562769371,
  exp: 1579478400,
  claim: {
    'Serto ID': {
      name: 'Sarah Adamson',
      dateOfBirth: '22-01-75',
      country: 'USA',
      children: [
        {
          name: 'Bob',
          age: 4,
        },
        {
          name: 'Alice',
          age: 9,
        },
      ],
    },
  },
  vc: [],
}

const bankVerifiableCredential = {
  iss: 'Deutsche Bank',
  sub: 'did:ethr:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74',
  type: 'Credit Worthy',
  iat: 1562769371,
  exp: 1579478400,
  claim: {
    'Credit Worthy': {
      name: 'Alice Chainy',
      dateOfBirth: '22-01-75',
      country: 'China',
      approvedLimit: 30000000,
    },
  },
  vc: [],
}

// tslint:disable-next-line:no-var-requires
const avatar1 = require('../../assets/images/space-x-logo.jpg')

export default (props: NavigationScreen) => {
  const credentials = [
    {
      title: sertoVerifiableCredential.type,
      issuer: sertoVerifiableCredential.iss,
      logo: avatar1,
      onPress: () =>
        props.navigation.navigate('Credential', {
          vc: sertoVerifiableCredential,
        }),
    },
    {
      title: bankVerifiableCredential.type,
      issuer: bankVerifiableCredential.iss,
      logo: avatar1,
      onPress: () =>
        props.navigation.navigate('Credential', {
          vc: bankVerifiableCredential,
        }),
    },
  ]
  const loadRequest = (requestData: any) => {
    setTimeout(() => {
      if (requestData.type === 'DISCLOSURE') {
        props.navigation.push('Request', {
          requestData,
        })
      }
    }, 500)
  }

  return (
    <Screen
      scrollEnabled
      fabButton={
        <Container>
          <FabButton
            testID={'SCAN_BTN'}
            onPress={() =>
              props.navigation.navigate('Scanner', {
                loadRequest,
              })
            }
            icon={Icons.SCAN}
          />
        </Container>
      }
    >
      <Container padding>
        <Container marginBottom>
          <Text type={Constants.TextTypes.H2} bold>
            Credentials
          </Text>
        </Container>
        <Container>
          {credentials.map((credential, index) => {
            return <Credential key={index} {...credential} />
          })}
        </Container>
      </Container>
    </Screen>
  )
}
