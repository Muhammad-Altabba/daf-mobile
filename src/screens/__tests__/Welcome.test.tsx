import 'react-native'
import React from 'react';
import Welcome from '../Welcome';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Welcome />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
