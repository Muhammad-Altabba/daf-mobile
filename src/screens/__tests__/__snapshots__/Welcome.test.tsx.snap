// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FF0000",
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to Serto!
  </Text>
</View>
`;
